pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -240,12 +240,16 @@ func (m *kubeGenericRuntimeManager) generateContainerConfig(container *v1.Contai
 	if err != nil {
 		return nil, cleanupAction, err
 	}
-
-	// Verify RunAsNonRoot. Non-root verification only supports numeric user.
-	if err := verifyRunAsNonRoot(pod, container, uid, username); err != nil {
-		return nil, cleanupAction, err
+	// https://github.com/kubernetes/enhancements/blob/master/keps/sig-windows/20190103-windows-node-support.md#v1container
+	// Windows does not have a root user, we cannot judge the root identity of the windows container by the way to judge the root of the linux container
+	// SecurityContext.RunAsUser, and SecurityContext.RunAsNonRoot These are Linux-specific attributes and cannot be used on windows containers
+	// Therefore, the judgment logic here does not apply to windows container
```

Comment:
would it be helpful to report an error back that configuration may be incorrect? Not necessarily by failing this verification, but maybe with the log message?

Reply:
Both windows container and linux will cover this piece of code, but it only makes sense to execute on linux. Therefore, I feel that there is no need to output some confusing logs on windows

Reply:
why do you think it will be confusing? If these settings were set by user, it may be important to know that settings were ignored. Or the assumption here that those log messages will never be seen by the person who set these configuration settings?

Reply:
The processing logic of `verifyRunAsNonRoot` is not suitable for windows containers. Many of the judgment fields are based on Linux containers, such as `SecurityContext.RunAsUser`, `image.uid`, `SecurityContext.RunAsNonRoot`. So I have circumvented this logic~ Maybe Windows container should be judged separately and used for user input check. Is this better? Do you think?

Reply:
I think it would be cleaner to split it out and judge differently.  Could use file suffix for the implementation as is done for [applyPlatformSpecificContainerConfig](https://github.com/kubernetes/kubernetes/blob/ee53488f19ff2fdd94cf311662e8a62845162871/pkg/kubelet/kuberuntime/kuberuntime_container.go#L277) in the [windows](https://github.com/kubernetes/kubernetes/blob/442a69c3bdf6fe8e525b05887e57d89db1e2f3a5/pkg/kubelet/kuberuntime/kuberuntime_container_windows.go#L37) and [linux](https://github.com/kubernetes/kubernetes/blob/442a69c3bdf6fe8e525b05887e57d89db1e2f3a5/pkg/kubelet/kuberuntime/kuberuntime_container_linux.go#L36) files.  For windows this can be a noop and if later there are privileged containers we would be able to add logic cleanly.